test/test_symbol_namespace.py

```diff
@@ -0,0 +1,46 @@
+"""
+Checks that the all exported symbols are properly namespaced, i.e., all start with "PQCLEAN_SCHEMENAME_".
+"""
+
+import os
+import pqclean
+import helpers
+import sys
+import unittest
+
+def test_symbol_namespace():
+    if sys.platform not in ['linux', 'darwin']: raise unittest.SkipTest()
+    for scheme in pqclean.Scheme.all_schemes():
+        for implementation in scheme.implementations:
+            yield check_symbol_namespace, scheme.name, implementation.name
+
+def check_symbol_namespace(scheme_name, implementation_name):
+    implementation = pqclean.Implementation.by_name(scheme_name, implementation_name)
+    helpers.run_subprocess(
+        ['make'],
+        implementation.path()
+    )
+    out = helpers.run_subprocess(
+        ['nm', '-g', 'lib{}_{}.a'.format(scheme_name, implementation_name)],
+        implementation.path() 
+    )
+
+    lines = out.strip().split("\n")
+    symbols = []
+    for line in lines:
```

Comment:
pythonic way would be:
```
symbols = list(itertools.filterfalse(lambda x: not (' T ' in x or ' D ' in x or ' S ' in x) , lines))
```

Reply:
This could remain an iterator and be moved into line 36, but let's do that some other time.